### PR TITLE
feat: Support a list of new AWS resources for AWS Python SDK

### PR DESCRIPTION
**Pull request by yiyuan-he**
### *Description of changes:*

These changes add auto-instrumentation support for the following AWS resources. Additionally, a new attribute for `aws.remote.resource.cfn.primary.identifier` is also added for each new resource.

- Populate `aws.sns.topic.arn` in Span by extracting `TopicArn` from the request body.
  - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-topic.html
  - The CFN Id should be the ARN of the Topic.
- Populate `aws.secretsmanager.secret.arn` in Span by extracting `ARN` from response body.
  - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-secretsmanager-secret.html
  - The CFN Id should be the ARN of the Secret.
- Populate `aws.stepfunctions.state_machine.arn` in Span by extracting `stateMachineArn` from the request body.
  - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html
  - The CFN Id should be the ARN of the State Machine.
- Populate `aws.stepfunctions.activity.arn` in Span by extracting `activityArn` from the request body.
  - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-activity.html
  - The CFN Id should be the ARN of the Activity.
- Populate `aws.lambda.function.name` in Span by extracting `FunctionName` from the request body.
  - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html
  - The CFN Id should be the ARN of the Lambda Function.
  - We populate `aws.lambda.function.arn` in Span by extraction `FunctionArn` from the response body because
  users are allowed to pass in both the name and ARN in the request body to retrieve the Lambda Function.
- Populate `aws.lambda.resource_mapping.id` in Span by extracting `UUID` from the request body.
  - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html
  - The CFN Id should be the UUID of the Event Source Mapping.

### *Test Plan:*
Set up a client-server with auto-instrumentation to verify that the correct span data is being generated.

sns_topic_arn_span_data_verification
secretsmanager_secret_arn_span_data_verification
sfn_statemachine_arn_span_data_verification
sfn_activity_arn_span_data_verification
lambda_function_name_span_data_verification
lambda_event_source_mapping_uuid_span_data_verification

Unit Tests for Instrumentation Patches and AWS Metric Attribute Generators.

instrumentation_patch_unit_test_verification
aws_metric_attribute_generator_unit_test_verification

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py [hunk 1]
```diff
@@ -309,6 +321,9 @@ def _normalize_remote_service_name(span: ReadableSpan, service_name: str) -> str
             "Bedrock Agent": _NORMALIZED_BEDROCK_SERVICE_NAME,
             "Bedrock Agent Runtime": _NORMALIZED_BEDROCK_SERVICE_NAME,
             "Bedrock Runtime": _NORMALIZED_BEDROCK_RUNTIME_SERVICE_NAME,
+            "Secrets Manager": _NORMALIZED_SECRETSMANAGER_SERVICE_NAME,
+            "SNS": _NORMALIZED_SNS_SERVICE_NAME,
+            "SFN": _NORMALIZED_STEPFUNCTIONS_SERVICE_NAME,
```
Comment (mxiamxia) [inline aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py:326]: missed Lambda?

File: aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py [hunk 2]
```diff
@@ -416,6 +434,25 @@ def _set_remote_type_and_identifier(span: ReadableSpan, attributes: BoundedAttri
         elif is_key_present(span, GEN_AI_REQUEST_MODEL):
             remote_resource_type = _NORMALIZED_BEDROCK_SERVICE_NAME + "::Model"
             remote_resource_identifier = _escape_delimiters(span.attributes.get(GEN_AI_REQUEST_MODEL))
+        elif is_key_present(span, AWS_SECRETSMANAGER_SECRET_ARN):
+            remote_resource_type = _NORMALIZED_SECRETSMANAGER_SERVICE_NAME + "::Secret"
+            remote_resource_identifier = _escape_delimiters(span.attributes.get(AWS_SECRETSMANAGER_SECRET_ARN))
```
Comment (mxiamxia) [inline aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py:439]: `remote_resource_identifier` should be resource name extracted from ARN string and `cloudformation_primary_identifier` should be the full ARN. :)

Comment (mxiamxia) [reply aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py:439]: the same should be applied to the other new resources added.

Comment (yiyuan-he) [reply aws-opentelemetry-distro/src/amazon/opentelemetry/distro/_aws_metric_attribute_generator.py]: Pushed some changes for this and updated screenshots in the description to show the new span data.

**Comment by yiyuan-he** [comment]
Synced offline with @mxiamxia and removed support for Kinesis Consumer Stream. It seems there is still some ambiguity regarding this resource that we need to resolve before we can support it.

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**